TargetComposition/SymBuilder/ApiProvider.cpp

```diff
@@ -1325,12 +1497,66 @@ void FieldObject::SetOffset(_In_ const Object& /*fieldObject*/, _In_ ComPtr<Fiel
     }
     else if (spFieldSymbol->InternalIsConstantValue())
     {
-        throw std::runtime_error("cannot set field offset of a field which is constnat value");
+        throw std::runtime_error("cannot set field offset of a field which is constant value");
     }
 
     CheckHr(spFieldSymbol->InternalSetSymbolOffset(fieldOffset));
 }
 
+std::optional<Object> FieldObject::GetValue(_In_ const Object& /*fieldObject*/, _In_ ComPtr<FieldSymbol>& spFieldSymbol)
+{
+    std::optional<Object> value;
+
+    if (spFieldSymbol->InternalIsConstantValue())
+    {
+        VARIANT const& val = spFieldSymbol->InternalGetSymbolValue();
+
+        //
+        // Unfortunately, DbgModelClientEx.h does not provide a "create this from a variant" so we need
+        // to step down one level.
+        //
+        ComPtr<IModelObject> spVal;
+        CheckHr(GetManager()->CreateIntrinsicObject(ObjectIntrinsic, const_cast<VARIANT *>(&val), &spVal));
+        value = spVal;
+    }
+    return value;
+}
+
+void FieldObject::SetValue(_In_ const Object& /*fieldObject*/, _In_ ComPtr<FieldSymbol>& spFieldSymbol,
+                           _In_ Object fieldValue)
+{
+    if (!spFieldSymbol->InternalIsConstantValue())
+    {
+        throw std::runtime_error("cannot set field value of a field which is not of constant value");
+    }
+    
+    if (fieldValue.GetKind() != ObjectIntrinsic)
+    {
+        throw std::invalid_argument("illegal value for constant field");
+    }
+
+    VARIANT vtVal;

```

Comment:
Do you need to clear this variant?

Reply:
I meant after you read its value and use it, do you need to clear it before exit the scope?

Reply:
Ok. I see you do that at the default.